src/dataEntities/startStop.ts

```diff
@@ -18,13 +19,20 @@ export abstract class StartStopService extends EventEmitter {
      */
     public static readonly STOPPED_EVENT = "stopped";
 
+    /**
+     * Each service has its own Logger instance
+     */
+    protected logger: Logger;
+
     /**
      * A service that requires starting and stopping.
      * Whoever constructs this service must start it before using it.
      * Whoever constructs this service must stop it after using it.
      */
     protected constructor(protected readonly name: string) {
         super();
+
+        this.logger = createNamedLogger(name);
```

Comment:
The names currently being used are of inconsistent formats. I think we should standardise them - documentation is enough. Something along the lines of "Names must be lower case, words separated by hypthons".

We can do a quick run time check that this is the case, and throw a configuration if necessary.

The problem with using the names as they are now is that they have spaces in them - which isnt nice for file names.

Reply:
Done, using the regexp `/^[a-z0-9\-]+$/`, so letters, numbers and hyphens.

Reply:
It's good feedback to include the offending name as part of the error message.

Reply:
Added better feedback in b083c9975182a41a572013f1993a80357e4fb63f.